index.d.ts

```diff
@@ -162,7 +162,7 @@ type SectionGridAllowedProps<ItemType = any> = Omit<SectionListProps<ItemType>,
 
 export interface SectionGridProps<ItemType = any>
   extends SectionGridAllowedProps<ItemType>, CommonProps<ItemType> {
-   sections: SectionItem<ItemType>[];
+   sections: readonly SectionItem<ItemType>[];
```

Comment:
How can this and `SectionItem` values be readonly? They are to be set by the user right? Did you test both SectionGrid and FlatGrid yourself after making this change?

Btw, did you face any issue when accidentally setting any value?

Reply:
The `readonly` type modifier only indicates that the array or object cannot be mutated after it's created, not necessarily that the array or object themselves are immutable. Adding `readonly` ensures the reference to the array/object cannot be reassigned or mutated. 

I did test both with this change and it works as expected. I didn't face any breaking errors, just typing errors -- I believe it's just generally recommended to treat props as immutable for better predictability and maintainability of components.